### ISSUE TITLE
[windows] run fail at other computer

### ISSUE THREAD

**Issue by ruizhang81** [google/flutter-desktop-embedding#291]
1、build success and run success in windows10
2、but move the  flutter-desktop-embedding\example\build\windows_fde\x64\Release\GLFW Example 
folder to other computer(win7 64)
3、run GLFW Example.exe 
4、show dialog "miss msvcp140d.dll..."
5、i add win10 msvcp140d.dll to GLFW Example folder 
6、show dialog "miss ***.dll..."
...

Is there a way to generate a directly run exe？

**Comment by stuartmorgan** [comment]
From a little bit of investigation, I believe the issue here is that the GN build on Windows is always linking to the debug libraries. The GN build will need to be updated to have Debug and Release modes that adjust the link flags.

For the library, this will likely be handled automatically during the migration to flutter/engine, but it should be fixed here for the plugins.

**Comment by ruizhang81** [comment]
modify build\win\toolchain\BUILD.gn
![image](https://user-images.githubusercontent.com/3723124/53782150-fb6d0f80-3f46-11e9-9d73-378f289127b1.png)

then run 
> $ tools\gn_dart gen out
> $ ninja -C out 

get the error
![image](https://user-images.githubusercontent.com/3723124/53782200-4424c880-3f47-11e9-962e-5714d810e377.png)

**Comment by stuartmorgan** [comment]
I'm not sure why you've switched to static runtime as part of that change; that may be part of the issue. But more generally, I would expect that a release mode would potentially involve other changes, not just that one option; certainly the Visual Studio example project changes at least a few flags.

I'm not familiar enough with the VS compiler to be able to suggest specific changes offhand. Someone with Windows toolchain experience will need to take a look, or it'll need to wait until I have more time to dig into the VS projects and compare.

**Comment by ruizhang81** [comment]
i want run it on other computer, dynamic compile will lose some dll file

**Comment by stuartmorgan** [comment]
> dynamic compile will lose some dll file

I'm not sure what this means exactly. The machine you're running it on will just need the dynamic runtime libraries, presumably.

Given that the prebuilt GLFW static library we're using currently works with the project, I assume that means that it's built with /MD, so if you want to use /MT from the whole project you'll need to start with writing rules to build GLFW from source.

**Comment by ruizhang81** [comment]
after build GLFW from source ,and use /MT
it work success on other computer,thanks

**Comment by stuartmorgan** [comment]
The fact that the build is hard-coded to use debug libraries is still an issue even if you've worked around it locally; reopening.

**Comment by stuartmorgan** [comment]
Blocked on https://github.com/flutter/flutter/issues/30672; once there's a release DLL, the example app can be updated to use it in release mode.

We'll need to change the plugins as well (either adding GN toolchain support for an actual release build mode, or not using GN for Windows plugins)

**Comment by boyen747** [comment]
> after build GLFW from source ,and use /MT
> it work success on other computer,thanks

Can you run on XP?

**Comment by stuartmorgan** [comment]
> Can you run on XP?

Keep in mind that the current shell is temporary placeholder, as described here:
https://github.com/flutter/flutter/wiki/Desktop-shells#windows

Even if it happens to run on XP (I have no idea if it does; that's certainly not a goal for it), the real shell will not.

**Comment by stuartmorgan** [comment]
Closing, as there's nothing FDE-specific left to track here. Once there's a release library for Windows the existing tooling integration would handle that from the Flutter side, and GN is no longer being used (so plugin project default configurations should be fine).

**Comment by Slihao** [comment]
> 从源构建GLFW，并使用/MT
> 它在其他计算机上工作成功，谢谢

I want the program to run on other computers what should I do (miss msvcp140d. DLL, little white, urgent)
